R/sparkR-mosaic/build_r_package.R

```diff
@@ -1,13 +1,37 @@
-repo<-"https://cran.ma.imperial.ac.uk/"
+repos = c(
+  "https://cran.ma.imperial.ac.uk" = "https://cran.ma.imperial.ac.uk"
+  ,"https://www.stats.bris.ac.uk/R" = "https://www.stats.bris.ac.uk/R"
+  ,"https://cran.rstudio.com/"  = "https://cran.rstudio.com/" 
+)
+
+mirror_is_up <- function(x){
+  out <- tryCatch({
+    available.packages(contrib.url(x))
+  }
+  ,error = function(cond){return(0)}
+  ,warning = function(cond){return(0)}
+  ,finally = function(cond){}
+  )
+  return(length(out))
+}
+
+mirror_status = lapply(repos, mirror_is_up)
+for(repo in names(mirror_status)){
+  if (mirror_status[[repo]] > 1){
+    repo <<- repo
+    break
+  }
+}
 
-install.packages("devtools", repos=repo)
+install.packages("pkgbuild", repos=repo)
 install.packages("roxygen2", repos=repo)
-devtools::install_github("apache/spark@v3.2.1", subdir='R/pkg')
 
-library(devtools)
+spark_location <- "/usr/spark-download/unzipped/spark-3.2.1-bin-hadoop2.7"
```

Comment:
this has hardcoded version in the path, this should be parametrised.
At some point we will move away from spark-3.2.1

Reply:
fwiw, i tried setting an env variable with the shell in the gh action and it broke, hence the hardcoding :)

Reply:
It would be better to parameterise this, but this is already a step in the right direction.
So I would merge it, and fix the tech debt when we upgrade to the next version